### PR TITLE
[xd |xprt] fix memory handling and refcounting. (reopened)

### PR DESCRIPTION
**Pull request by sswen**
Recreated PR

### CODE REVIEW

**Review by was4 — COMMENTED** [review]
rename. this commit title doesn't tell us anything

File: ntirpc/rpc/svc.h [hunk 1]
```diff
@@ -155,6 +155,7 @@ typedef struct svc_init_params {
 #define SVC_XPRT_FLAG_DESTROYED		0x0020	/* SVC_DESTROY() was called */
 #define SVC_XPRT_FLAG_DESTROYING	0x0040	/* (*xp_destroy) was called */
 #define SVC_XPRT_FLAG_VSOCK             0x0080
+#define SVC_XPRT_FLAG_ALLOCATED		0x8000
```
Comment (was4) [inline ntirpc/rpc/svc.h:158]: Need better name. _ALLOCATED doesn't tell us anything. Perhaps _VC_DATA.

**Review by was4 — COMMENTED** [review]
Need more descriptive title (duplicate). Also, more text in commit body describing the purpose and bug fixes.

File: ntirpc/rpc/svc.h (hunk 1 above)
Comment (was4) [reply ntirpc/rpc/svc.h:158]: Belay that. I don't see why this flag is needed. If the pointer is NULL, it isn't allocated.

Comment (sswen) [reply ntirpc/rpc/svc.h:158]: No. The way the code is written we need a way to distinguish between memory which is allocated just now and memory which was already available and therefore already got initialized.

**Comment by sswen** [comment]
Updated both commit messages.

**Review by was4 — COMMENTED** [review]
This one is now committed. With credit

Rename svc_vc_ncreatef 

Use SVC_CREATE_FLAG_CLOSE (SVC_XPRT_FLAG_CLOSE).
Inspired-by: Swen Schillig <swen@vnet.ibm.com>

**Review by was4 — COMMENTED** [review]
This one is now partially committed, although completely revised:

Replace rpc_dplx_lookup_rec with svc_xprt_lookup 

New SVC_XPRT_FLAG_INITIALIZED prevents repeated initialization.

**Comment by ffilz** [comment]
I think we can close this one?

**Pull request closed without merging**